src/api/firebase.js

```diff
@@ -173,3 +173,71 @@ export async function deleteItem() {
 	 * this function must accept!
 	 */
 }
+
+/**
+ * Compare purchase history of list items and sort items by three attributes:
+ * 1) Purchase status, 2) Urgency, and 3) Alphabetically
+ * add a property to each item indicating the urgency of the next purchase
+ * @param {object} data The raw list data to be sorted.
+ */
+export function comparePurchaseUrgency(data) {
+	// Create an empty, nested array for sorted data categories
+	const sortedData = [];
+	for (let i = 0; i < 6; i++) {
+		sortedData.push([]);
+	}
+
+	// Loop through all items and assign urgency based on days until next purchase
+	for (let item of data) {
+		// If the item has been purchased before, return days since last purchase
+		// If the item has never been purchased, return the days since item creation
+		const daysSincePurchase = item.dateLastPurchased
+			? getDaysBetweenDates(item.dateLastPurchased)
+			: getDaysBetweenDates(item.dateCreated);
+
+		// Assign urgency
+		if (daysSincePurchase >= 60) {
```

Comment:
I'm seeing a lot of `magic numbers` floating around here. Let's revive the use of [`numDaysInEstimate`](https://github.com/the-collab-lab/tcl-47-smart-shopping-list/blob/5392e049b46fce5f7a1cc0a535fa10a443cd202c/src/views/AddItem.jsx#L6) and add the new `inactive` and `overdue` categories to it.

Reply:
I went around and around on this. I tried replacing the numbers with variable names and in the end that got so complex/confusing that I ended up adding comments to explain the numbers. For learning, if you have a clean way to handle this in mind, I'd love to talk about it again.